### ISSUE TITLE
Syntax highlighting breaks with nested sql.array

### ISSUE THREAD

**Issue by LorenzHenk** [thebearingedge/vscode-sql-lit#9]
Hey, thanks for this awesome extension!

I've noticed that sql highlighting fails to properly highlight if there's e.g. an `sql.array` or `sql.join` within an sql template string

<img width="551" alt="image" src="https://user-images.githubusercontent.com/10603631/220164939-18a9c6bb-fce9-4107-b948-65606d7e1956.png">

failing code example:

```ts
export function fooFailing() {
  return sql`SELECT (${sql.array([1, 2, 3], 'int8')})`;
}
```

**Comment by thebearingedge** [comment]
Hi @LorenzHenk thanks for the report.

I do see that in `fooFailing`, the square brackets are green and not brown. Is there another problem I do not see?

Related: When trying `fooFailing` with Monokai Dark Soda, I see a different inconsistency. The closing curly brace of the interpolation does not match.

![code2](https://user-images.githubusercontent.com/7432943/220733969-755ad426-9567-4ed8-ad8d-627944efb5bb.png)

![code](https://user-images.githubusercontent.com/7432943/220734019-e5de2414-eae0-4ac1-9723-cae790d3c017.png)

When you have time, can you post a larger breakage if possible?

**Comment by LorenzHenk** [comment]
In my initial screenshot you can see that the curly braces of the `fooFailing` function body are red.

In the screenshot below you can see that the syntax highlighting for the `export async` keywords are broken in the function that is defined below `fooFailing`.


<img width="549" alt="image" src="https://user-images.githubusercontent.com/10603631/220929598-ca9c26b1-c5e9-481e-b593-9e01e294e155.png">


It becomes even worse when using a normal template string after the nested `sql.array`, in this example in the `console.log` statement:

<img width="502" alt="image" src="https://user-images.githubusercontent.com/10603631/220941577-880fc7da-6c6b-473a-b53a-4f0f27296a00.png">

You can see that the code below appears highlighted as if it was text within a string.

**Comment by thebearingedge** [comment]
Ok, thank you very much!

**Comment by thebearingedge** [comment]
Hi @LorenzHenk I believe this is fixed in version 0.6.1

https://open-vsx.org/extension/thebearingedge/vscode-sql-lit

**Comment by LorenzHenk** [comment]
It works, thanks a lot! 🚀
